pkg/doc/signature/suite/bbsblssignatureproof2020/signer.go

```diff
@@ -45,44 +46,65 @@ func (s *Suite) SelectiveDisclosure(doc map[string]interface{}, revealDoc map[st
 
 	delete(docCompacted, "proof")
 
-	proofInfo, err := getBlsProof(rawProofs)
+	blsSignatures, err := getBlsProofs(rawProofs)
 	if err != nil {
 		return nil, err
 	}
 
-	blsProof := proofInfo.blsProof
+	var revealDocumentResult map[string]interface{}
+
+	proofs := make([]map[string]interface{}, len(blsSignatures))
+
+	for i, blsProof := range blsSignatures {
+		revealedDoc, derivedProof, dErr := generateSignatureProof(blsProof, resolver, docCompacted,
```

Comment:
can the revealedDoc from the first iteration be reused for the rest?

Reply:
Reason I ask is because these JSON-LD operations seem expensive from my experience. Especially considering that there are probably network calls being made.

I suspect JSON-LD is one of the reasons why our zcapld implementation is slow.

Reply:
I got it. Great input, thank you! Let me change the program flow in this regard.

Reply:
Refactored.